lookout/style/format/benchmarks/top_repos_quality.py

```diff
@@ -0,0 +1,250 @@
+    for line in report.split("\n"):
+        if "weighted avg" in line:
+            _, prec, recall, f1, support = line.split("` | `")
+            prec, recall, f1, support = [float(f.replace("`", "").replace("|", "")) for f in
+                                              (prec, recall, f1, support)]
+            return prec, recall, f1, int(support)
+
+
+def get_model_summary(report):
+    """Extract model summary - number of rules and avg. len."""
+    pattern = " rules, avg.len. "
+    for line in report.split("\n"):
+        if pattern in line:
+            line = line.replace("`", "")
+            res = list(map(float, line.split(pattern)))
+            assert len(res) == 2
+            n_rules, avg_len = res
+            n_rules = int(n_rules)
+    return n_rules, avg_len
+
+
+def main(args):
+    """Generate report for each repository & calculate average results for repositories."""
+    fetch()  # download executable
+    # prepare output directory
+    dir_to_save = args.output
+    if not os.path.exists(dir_to_save):
+        os.mkdir(dir_to_save)
+    assert os.path.isdir(dir_to_save), "Output should be a directory"
+    reports = []
+    for repo in REPOSITORIES:
+        repo, to_commit, from_commit = repo.split()
+        report_loc = os.path.join(dir_to_save, get_repo_name(repo))
+        quality_rep_loc = report_loc + ".quality_report.md"
+        model_rep_loc = report_loc + ".model_report.md"
+        # generate or read report
+        if args.force or not os.path.exists(quality_rep_loc) or not os.path.exists(model_rep_loc):
+            # Skip this step if report was already generated
+            quality_report, model_report = measure_quality(repo, to_commit=to_commit,
+                                                           from_commit=from_commit,
+                                                           bblfsh=args.bblfsh_address)
+
+            with open(quality_rep_loc, "w", encoding="utf-8") as f:
+                f.write(quality_report)
+            with open(model_rep_loc, "w", encoding="utf-8") as f:
+                f.write(model_report)
+        else:
+            with open(quality_rep_loc, encoding="utf-8") as f:
+                quality_report = f.read()
+            with open(model_rep_loc, encoding="utf-8") as f:
+                model_report = f.read()
+        report = "\n".join((quality_report, model_report))
+        reports.append((repo, report))
+    # precision, recall, f1, support, n_rules, avg_len stats
+    agg = []
+    table = []
+    with io.StringIO() as output:
+        table.append(("repo", "prec", "recall", "f1", "support", "n_rules", "avg_len"))
+        for repo, report in reports:
+            precision, recall, f1, support = get_presicion_recall_f1_support(report)
+            try:
+                n_rules, avg_len = get_model_summary(report)
+            except Exception as e:
+                print(e)
```

Comment:
Should be handled better :) I guess we should print a warning and continue; break into `pdb` if debug mode

Reply:
it's debugging part, somehow missed it before PR